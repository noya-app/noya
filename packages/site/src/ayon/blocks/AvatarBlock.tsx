import { applyCommonProps } from '@noya-design-system/protocol';
import { BlockDefinition } from 'noya-state';
import { isExternalUrl } from 'noya-utils';
import { getBlockClassName } from '../tailwind/tailwind';
import { getParameters } from '../utils/getMappedParameters';
import { isApproximatelySquare, isWithinRectRange } from './score';
import { avatarSymbolId } from './symbolIds';
import { avatarSymbol } from './symbols';

const AVATAR_SIZES = ['2xs', 'xs', 'sm', 'md', 'lg', 'xl', '2xl'];

export const AvatarBlock: BlockDefinition = {
  symbol: avatarSymbol,
  infer: ({ frame, blockText }) =>
    isWithinRectRange({
      rect: frame,
      minWidth: 30,
      minHeight: 30,
      maxWidth: 120,
      maxHeight: 120,
    }) && isApproximatelySquare(frame, 0.2)
      ? 0.8
      : 0,
  render: ({ h, Components: { [avatarSymbolId]: Avatar } }, props) => {
    const parameters = getParameters(props.blockParameters);
    const content = props.blockText ?? '';
    const src = isExternalUrl(content) ? content : undefined;
    const name = !src ? content : undefined;
    const hashtags = Object.keys(parameters);

    // Size up in 14px increments
    const size = props.frame
      ? AVATAR_SIZES[
          Math.min(
            Math.floor(Math.max(props.frame.width, props.frame.height) / 14),
            AVATAR_SIZES.length - 1,
          )
        ]
      : 'md';

    const sizePx = props.frame
      ? Math.min(props.frame.width, props.frame.height)
      : undefined;

    const widthHashtag = hashtags.find((hashtag) => hashtag.startsWith('w-'));
    const widthMatch = widthHashtag ? widthHashtag.match(/w-(\d+)/) : undefined;
    const width = widthMatch ? Number(widthMatch[1]) : undefined;
    const tailwindSize = width ? `${width * 0.25}rem` : undefined;

    return h(Avatar, {
      ...applyCommonProps(props),
      size,
      name,
      src,
      style: {
        aspectRatio: '1',
        ...(sizePx && {
          width: `${sizePx}px`,
          height: `${sizePx}px`,
        }),
        ...(tailwindSize && {
          width: tailwindSize,
          height: tailwindSize,
        }),
      },
      className: getBlockClassName(hashtags),
    });
  },
};
