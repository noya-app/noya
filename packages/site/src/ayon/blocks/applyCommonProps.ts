import { CommonProps } from '@noya-design-system/protocol';
import { BlockProps } from 'noya-state';

export function applyCommonProps(
  props: BlockProps,
): Pick<CommonProps, '_passthrough'> {
  return {
    _passthrough: {
      ...(props.dataSet && {
        key: props.dataSet.id,
        'data-noya-id': props.dataSet.id,
        'data-noya-parent-id': props.dataSet.parentId,
      }),
    },
  };
}
