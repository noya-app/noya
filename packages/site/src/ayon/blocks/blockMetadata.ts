import { Size } from 'noya-geometry';
import {
  avatarSymbolId,
  boxSymbolId,
  buttonSymbolId,
  cardSymbolId,
  checkboxSymbolId,
  featureItemSymbolId,
  headerBarSymbolId,
  heading1SymbolId,
  heading2SymbolId,
  heading3SymbolId,
  heading4SymbolId,
  heading5SymbolId,
  heading6SymbolId,
  heroSymbolId,
  heroWithImageSymbolId,
  iconSymbolId,
  imageSymbolId,
  inputSymbolId,
  linkSymbolId,
  radioSymbolId,
  selectSymbolId,
  sidebarSymbolId,
  signInSymbolId,
  switchSymbolId,
  tableSymbolId,
  tagSymbolId,
  textareaSymbolId,
  textSymbolId,
  tileCardSymbolId,
  writeSymbolId,
} from '../symbols/symbolIds';

export type BlockCategory = 'application' | 'marketing' | 'element';

export type PreferredOverride = {
  symbolId: string;
  blockText: string;
  resolvedBlockText?: string;
};

export type BlockMetadata = {
  name: string;
  category: BlockCategory;
  preferredSize: Size;
  preferredBlockText?: string;
  preferredResolvedBlockText?: string;
  preferredOverrides?: PreferredOverride[];
  hideInDocs?: boolean;
};

const headingMetadata: BlockMetadata = {
  name: 'Heading',
  category: 'element',
  preferredSize: { width: 700, height: 100 },
  preferredBlockText: 'All About Cats',
};

const headingNMetadata: BlockMetadata = {
  ...headingMetadata,
  hideInDocs: true,
};

// This is in a separate file so docs don't have to import Chakra components directly.
// Docs can import this synchronously and import(Ayon)
export const blockMetadata: Record<string, BlockMetadata> = {
  [avatarSymbolId]: {
    name: 'Avatar',
    category: 'element',
    preferredSize: { width: 64, height: 64 },
  },
  [buttonSymbolId]: {
    name: 'Button',
    category: 'element',
    preferredSize: { width: 200, height: 40 },
  },
  [linkSymbolId]: {
    name: 'Link',
    category: 'element',
    preferredSize: { width: 200, height: 40 },
  },
  [boxSymbolId]: {
    name: 'Box',
    category: 'element',
    preferredSize: { width: 200, height: 200 },
  },
  [heroSymbolId]: {
    name: 'Hero',
    category: 'marketing',
    preferredSize: { width: 600, height: 400 },
  },
  [heroWithImageSymbolId]: {
    name: 'Hero with Image',
    category: 'marketing',
    preferredSize: { width: 1280, height: 720 },
  },
  [tagSymbolId]: {
    name: 'Tag',
    category: 'element',
    preferredSize: { width: 140, height: 30 },
  },
  [cardSymbolId]: {
    name: 'Card',
    category: 'marketing',
    preferredSize: { width: 250, height: 300 },
    preferredOverrides: [
      {
        symbolId: imageSymbolId,
        blockText: 'landscape',
        resolvedBlockText:
          'https://images.unsplash.com/photo-1514917860136-ee8b88e8c9c9?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=300&ixid=MnwxfDB8MXxyYW5kb218MHx8bGFuZHNjYXBlLGJ1c3Rlcnx8fHx8fDE2Nzc2NDIxNDM&ixlib=rb-4.0.3&q=80&utm_campaign=api-credit&utm_medium=referral&utm_source=unsplash_source&w=300',
      },
    ],
  },
  [tileCardSymbolId]: {
    name: 'Tile Card',
    category: 'marketing',
    preferredSize: { width: 250, height: 250 },
  },
  [iconSymbolId]: {
    name: 'Icon',
    category: 'element',
    preferredSize: { width: 48, height: 48 },
  },
  [imageSymbolId]: {
    name: 'Image',
    category: 'element',
    preferredSize: { width: 300, height: 300 },
    preferredBlockText: 'landscape',
    preferredResolvedBlockText:
      'https://images.unsplash.com/photo-1514917860136-ee8b88e8c9c9?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=300&ixid=MnwxfDB8MXxyYW5kb218MHx8bGFuZHNjYXBlLGJ1c3Rlcnx8fHx8fDE2Nzc2NDIxNDM&ixlib=rb-4.0.3&q=80&utm_campaign=api-credit&utm_medium=referral&utm_source=unsplash_source&w=300',
  },
  [writeSymbolId]: {
    name: 'Write',
    category: 'element',
    preferredSize: { width: 300, height: 300 },
    preferredBlockText: 'Cats are great pets',
    preferredResolvedBlockText:
      'Cats are incredibly intelligent creatures with personalities all of their own. They make loyal and loving companions who can provide years of fun and joy. Cats come in all shapes, sizes, and colors and offer a variety of health benefits to their human owners. From their natural mouse-catching abilities to their playful antics, cats make terrific pet choices for a wide variety of households.',
  },
  [textSymbolId]: {
    name: 'Text',
    category: 'element',
    preferredSize: { width: 300, height: 150 },
    preferredBlockText:
      'Cats are great pets. They make loyal and loving companions who can provide years of fun and joy.',
  },
  [heading1SymbolId]: headingMetadata,
  [heading2SymbolId]: headingNMetadata,
  [heading3SymbolId]: headingNMetadata,
  [heading4SymbolId]: headingNMetadata,
  [heading5SymbolId]: headingNMetadata,
  [heading6SymbolId]: headingNMetadata,
  [signInSymbolId]: {
    name: 'Sign In',
    category: 'application',
    preferredSize: { width: 350, height: 250 },
  },
  [tableSymbolId]: {
    name: 'Table',
    category: 'application',
    preferredSize: { width: 600, height: 400 },
  },
  [sidebarSymbolId]: {
    name: 'Sidebar',
    category: 'application',
    preferredSize: { width: 250, height: 600 },
  },
  [headerBarSymbolId]: {
    name: 'Header Bar',
    category: 'application',
    preferredSize: { width: 800, height: 80 },
  },
  [checkboxSymbolId]: {
    name: 'Checkbox',
    category: 'element',
    preferredSize: { width: 300, height: 30 },
  },
  [radioSymbolId]: {
    name: 'Radio',
    category: 'element',
    preferredSize: { width: 300, height: 30 },
  },
  [selectSymbolId]: {
    name: 'Select',
    category: 'element',
    preferredSize: { width: 300, height: 40 },
  },
  [textareaSymbolId]: {
    name: 'Textarea',
    category: 'element',
    preferredSize: { width: 240, height: 120 },
  },
  [inputSymbolId]: {
    name: 'Input',
    category: 'element',
    preferredSize: { width: 240, height: 40 },
  },
  [switchSymbolId]: {
    name: 'Switch',
    category: 'element',
    preferredSize: { width: 50, height: 35 },
  },
  [featureItemSymbolId]: {
    name: 'Feature Item',
    category: 'marketing',
    preferredSize: { width: 400, height: 160 },
  },
};
