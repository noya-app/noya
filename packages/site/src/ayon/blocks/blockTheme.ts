// Blue
export const accentColor = {
  50: '#eff6ff',
  100: '#dbeafe',
  200: '#bfdbfe',
  300: '#93c5fd',
  400: '#60a5fa',
  500: '#3b82f6',
  600: '#2563eb',
  700: '#1d4ed8',
  800: '#1e40af',
  900: '#1e3a8a',
};

// Slate
export const neutralColor = {
  50: '#f8fafc',
  100: '#f1f5f9',
  200: '#e2e8f0',
  300: '#cbd5e1',
  400: '#94a3b8',
  500: '#64748b',
  600: '#475569',
  700: '#334155',
  800: '#1e293b',
  900: '#0f172a',
};

export const buttonColors = {
  light: {
    backgroundColor: '#f1f5f9',
    color: '#000',
  },
  lightDisabled: {
    backgroundColor: '#e2e8f0',
    color: '#000',
  },
  dark: {
    backgroundColor: '#475569',
    color: '#fff',
  },
  darkDisabled: {
    backgroundColor: '#8799b2',
    color: '#fff',
  },
};
