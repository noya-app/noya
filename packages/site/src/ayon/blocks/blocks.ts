import Sketch from 'noya-file-format';
import {
  heroButtonRowSymbol,
  heroHeadlineStackSymbol,
  heroSymbol,
} from '../symbols/composed/HeroSymbol';
import { avatarSymbol } from '../symbols/primitive/AvatarSymbol';
import { boxSymbol } from '../symbols/primitive/BoxSymbol';
import { buttonSymbol } from '../symbols/primitive/ButtonSymbol';
import { checkboxSymbol } from '../symbols/primitive/CheckboxSymbol';
import { tagSymbol } from '../symbols/primitive/TagSymbol';
import { heroSymbolV1Id } from '../symbols/symbolIds';

export const allSymbols = [
  avatarSymbol,
  buttonSymbol,
  boxSymbol,
  heroSymbol,
  heroHeadlineStackSymbol,
  heroButtonRowSymbol,
  tagSymbol,
  checkboxSymbol,
];

export const allInsertableSymbols = allSymbols.filter(
  (symbol) => !symbol.blockDefinition?.isPassthrough,
);

export const symbolMap: Record<string, Sketch.SymbolMaster> = {
  ...Object.fromEntries(allSymbols.map((symbol) => [symbol.symbolID, symbol])),
  // Add extra mappings here (e.g. heroV1 => heroV2)
  [heroSymbolV1Id]: heroSymbol,
};
