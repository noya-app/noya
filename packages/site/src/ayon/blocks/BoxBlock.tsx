import { BlockDefinition } from 'noya-state';
import { findLast } from 'noya-utils';
import { parseBlock } from '../parse';
import { applyCommonProps } from './applyCommonProps';
import { accentColor } from './blockTheme';
import { getBlockThemeColors } from './colors';
import { boxSymbolId } from './symbolIds';
import { boxSymbol } from './symbols';
import {
  getLastClassInGroup,
  parametersToTailwindStyle,
  tailwindBlockClasses,
} from './tailwind';

export const BoxBlock: BlockDefinition = {
  symbol: boxSymbol,
  parser: 'regular',
  infer: ({ frame, blockText }) => 0.1,
  hashtags: ['left', 'center', 'right', 'dark', ...tailwindBlockClasses],
  render: ({ h, Components: { [boxSymbolId]: Box } }, props) => {
    const { content, parameters } = parseBlock(props.blockText, 'regular');
    const hashtags = Object.keys(parameters);

    const bgClass = getLastClassInGroup('background', hashtags);
    const dynamicClass = bgClass
      ? typeof parameters[bgClass] === 'string'
      : undefined;

    const backgroundColor = parameters.dark
      ? getBlockThemeColors({ dark: true, accent: false }).backgroundColor
      : bgClass
      ? dynamicClass
        ? parameters[bgClass]
        : undefined
      : [content]
          .concat(hashtags)
          .find((value) => CSS.supports('color', `${value}`)) ??
        accentColor[50];

    const { justify, items } = simpleFlex(hashtags);

    const style = parametersToTailwindStyle({
      ...parameters,
      ...(justify && { [justify]: true }),
      ...(items && { [items]: true }),
    });

    return h(
      Box,
      {
        ...applyCommonProps(props),
        style: {
          display: 'flex',
          ...style,
          backgroundColor,
          ...(props.frame && {
            width: `${props.frame.width}px`,
            height: `${props.frame.height}px`,
          }),
        },
      },
      props.children,
    );
  },
};

// Simplify flexbox alignment by handling row vs. column automatically
function simpleFlex(hashtags: string[]) {
  const flexKey = findLast(hashtags, (value) =>
    /^(flex-row|flex-col)$/.test(value),
  );
  const alignmentKey = findLast(hashtags, (value) =>
    /^(left|center|right)$/.test(value),
  );

  let justify: string | undefined;
  let items: string | undefined;

  if (flexKey === 'flex-row') {
    switch (alignmentKey) {
      case 'left':
        justify = 'justify-start';
        break;
      case 'center':
        justify = 'justify-center';
        break;
      case 'right':
        justify = 'justify-end';
        break;
    }

    if (hashtags.includes('center')) {
      items = 'items-center';
    }
  } else if (flexKey === 'flex-col') {
    switch (alignmentKey) {
      case 'left':
        items = 'items-start';
        break;
      case 'center':
        items = 'items-center';
        break;
      case 'right':
        items = 'items-end';
        break;
    }

    if (hashtags.includes('center')) {
      justify = 'justify-center';
    }
  }

  return { justify, items };
}
