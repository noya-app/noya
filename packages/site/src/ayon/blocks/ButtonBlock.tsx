import { BlockDefinition } from 'noya-state';
import { parseBlock } from '../parse';
import { buttonColors } from './blockTheme';
import { isWithinRectRange } from './score';
import { buttonSymbolId } from './symbolIds';
import { buttonSymbol } from './symbols';
import { getBlockClassName } from './tailwind';

const placeholderText = 'Submit';

const globalHashtags = [
  'light',
  'dark',
  'primary',
  'secondary',
  'warning',
  'danger',
  'disabled',
  'xs',
  'md',
  'lg',
];

const parser = 'regular';

type ButtonColorKey = keyof typeof buttonColors;

const colorsKeys = new Set<ButtonColorKey>([
  'light',
  'dark',
  'primary',
  'secondary',
  'warning',
  'danger',
]);

export const ButtonBlock: BlockDefinition = {
  symbol: buttonSymbol,
  parser,
  hashtags: globalHashtags,
  placeholderText,
  infer: ({ frame, blockText }) =>
    isWithinRectRange({
      rect: frame,
      minWidth: 60,
      minHeight: 30,
      maxWidth: 300,
      maxHeight: 80,
    })
      ? 0.8
      : 0,
  render: ({ h, Components: { [buttonSymbolId]: Button } }, props) => {
    const {
      content,
      parameters: { xs, lg, md, disabled, ...parameters },
    } = parseBlock(props.blockText, parser, { placeholder: placeholderText });

    const hashtags = Object.keys(parameters);

    const colorKey = hashtags
      .slice()
      .reverse()
      .find((key) => colorsKeys.has(key as ButtonColorKey)) as
      | ButtonColorKey
      | undefined;

    const buttonColorKey = colorKey ?? 'default';

    const colors = buttonColors[buttonColorKey];

    let size = xs ? 'xs' : lg ? 'lg' : md ? 'md' : undefined;

    if (props.frame && size === undefined) {
      if (props.frame.height < 30) {
        size = 'xs' as const;
      } else if (props.frame.height > 50) {
        size = 'lg' as const;
      } else {
        size = 'md' as const;
      }
    }

    return h(
      Button,
      {
        ...(props.dataSet && {
          key: props.dataSet.id,
          'data-noya-id': props.dataSet.id,
          'data-noya-parent-id': props.dataSet.parentId,
        }),
        disabled: !!disabled,
        style: {
          color: colors.color,
          backgroundColor: colors.backgroundColor,
          ...(props.frame && {
            width: `${props.frame.width}px`,
          }),
        },
        className: getBlockClassName(hashtags),
      },
      content,
    );
  },
};
