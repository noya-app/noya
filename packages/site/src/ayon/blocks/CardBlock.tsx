import { BlockDefinition } from 'noya-state';
import { BoxBlock } from './BoxBlock';
import { renderStack } from './render';
import { isWithinRectRange } from './score';
import { cardSymbol } from './symbols';

export const CardBlock: BlockDefinition = {
  editorVersion: 2,
  symbol: cardSymbol,
  hashtags: BoxBlock.hashtags,
  isComposedBlock: true,
  infer: ({ frame, blockText, siblingBlocks }) => {
    return Math.max(
      isWithinRectRange({
        rect: frame,
        minWidth: 200,
        minHeight: 250,
        maxWidth: 300,
        maxHeight: 400,
      })
        ? 1
        : 0,
      0.1,
    );
  },
  render: (env, props) => renderStack(env, { props, block: CardBlock }),
};
