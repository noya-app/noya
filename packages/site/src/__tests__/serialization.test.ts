import { readFileSync } from 'fs';
import { SketchModel } from 'noya-sketch-model';
import { Layers, Overrides } from 'noya-state';
import path from 'path';
import { Blocks } from '../ayon/blocks/blocks';
import {
  heading4SymbolId,
  heroSymbolV2Id,
  heroWithImageSymbolId,
} from '../ayon/blocks/symbolIds';
import {
  EditorBlockContent,
  extractBlockContent,
  fromContent,
  toContent,
} from '../ayon/editor/serialization';

// Jest doesn't know how to import a text file, so we mock it
jest.mock('../../safelist.txt', () => {
  return {
    default: readFileSync(path.join(__dirname, '../../safelist.txt'), 'utf8'),
  };
});

function filterEmptyOverrides(blockContent: EditorBlockContent) {
  return {
    ...blockContent,
    overrides: blockContent.overrides.filter((o) => o.value !== ''),
  };
}

test('serializes block', () => {
  const instance = SketchModel.symbolInstance({
    do_objectID: 'b',
    blockText: '',
    symbolID: heroSymbolV2Id,
    frame: SketchModel.rect({
      width: 400,
      height: 400,
    }),
  });

  const symbol = Blocks[instance.symbolID].symbol;

  const originalContent = extractBlockContent(instance);
  const nodes = fromContent(symbol, originalContent);
  const content = toContent(symbol, nodes);

  expect(filterEmptyOverrides(content)).toEqual(originalContent);
});

test('serializes block with layer text', () => {
  const instance = SketchModel.symbolInstance({
    do_objectID: 'a',
    symbolID: heroSymbolV2Id,
    frame: SketchModel.rect({
      width: 400,
      height: 400,
    }),
  });

  const symbol = Blocks[instance.symbolID].symbol;

  const targetLayer = symbol.layers
    .filter(Layers.isSymbolInstance)
    .find((layer) => layer.symbolID === heading4SymbolId);

  instance.overrideValues.push(
    SketchModel.overrideValue({
      overrideName: Overrides.encodeName(
        [targetLayer!.do_objectID],
        'blockText',
      ),
      value: 'hello',
    }),
  );

  const originalContent = extractBlockContent(instance);

  expect(originalContent).toMatchSnapshot('originalContent');

  const nodes = fromContent(symbol, originalContent);

  expect(nodes).toMatchSnapshot('nodes');

  const content = toContent(symbol, nodes);

  expect(content).toMatchSnapshot('content');

  expect(filterEmptyOverrides(content)).toEqual(originalContent);
  expect(nodes).toMatchSnapshot();
});

test('serializes nested block', () => {
  const instance = SketchModel.symbolInstance({
    do_objectID: 'a',
    symbolID: heroWithImageSymbolId,
    frame: SketchModel.rect({
      width: 400,
      height: 400,
    }),
  });

  const symbol = Blocks[instance.symbolID].symbol;

  const heroLayer = symbol.layers
    .filter(Layers.isSymbolInstance)
    .find((layer) => layer.symbolID === heroSymbolV2Id);

  const nestedSymbol = Blocks[heroLayer!.symbolID].symbol;

  const targetLayer = nestedSymbol.layers
    .filter(Layers.isSymbolInstance)
    .find((layer) => layer.symbolID === heading4SymbolId);

  instance.overrideValues.push(
    SketchModel.overrideValue({
      overrideName: Overrides.encodeName(
        [heroLayer!.do_objectID, targetLayer!.do_objectID],
        'blockText',
      ),
      value: 'hello',
    }),
  );

  const originalContent = extractBlockContent(instance);

  expect(originalContent).toEqual({
    blockText: '',
    overrides: [
      SketchModel.overrideValue({
        overrideName: Overrides.encodeName(
          [heroLayer!.do_objectID, targetLayer!.do_objectID],
          'blockText',
        ),
        value: 'hello',
      }),
    ],
    symbolId: symbol.symbolID,
    layerPath: [],
  });

  const nodes = fromContent(symbol, originalContent);
  const content = toContent(symbol, nodes);
  expect(filterEmptyOverrides(content)).toEqual(originalContent);
});
