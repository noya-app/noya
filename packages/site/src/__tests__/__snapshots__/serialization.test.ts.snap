// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serializes block with layer text 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "text": "",
      },
    ],
    "isRoot": true,
    "layerPath": Array [],
    "symbolId": "61eb943b-7ee7-4b33-9315-aac7cb2d2764",
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "text": "",
      },
    ],
    "isRoot": false,
    "layerPath": Array [
      "d6593501-b089-4390-bbe2-fb10afb5df5a",
      "1dea1c4d-f1bd-473b-a1aa-a0c6a1481ae2",
    ],
    "symbolId": "a1f7266f-50cc-416f-9ed2-4af4bca30257",
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "text": "",
      },
    ],
    "isRoot": false,
    "layerPath": Array [
      "d6593501-b089-4390-bbe2-fb10afb5df5a",
      "ef2d5b26-aa1c-40d3-8bab-37c10bccc5cb",
    ],
    "symbolId": "955f4cb1-7879-4331-a359-a21a031147cb",
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "text": "hello",
      },
    ],
    "isRoot": false,
    "layerPath": Array [
      "aa722c35-9ba4-4bf3-a5d0-f7d17f02c361",
    ],
    "symbolId": "b8441949-12ba-42be-8f87-5cf222af70d0",
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "text": "",
      },
    ],
    "isRoot": false,
    "layerPath": Array [
      "83d2fdeb-6f4d-4948-a677-fe1f2aac64d5",
      "6b386c69-d6cf-4c2f-ae06-c92af43268d5",
    ],
    "symbolId": "cdf2346b-cb21-4f23-8d93-7c4fb2e3a5a0",
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "text": "",
      },
    ],
    "isRoot": false,
    "layerPath": Array [
      "83d2fdeb-6f4d-4948-a677-fe1f2aac64d5",
      "eee85c94-7361-4bcf-8afb-f59c6e8661f7",
    ],
    "symbolId": "5be80b87-7bd6-4090-8947-316d54064544",
    "type": "paragraph",
  },
]
`;

exports[`serializes block with layer text: content 1`] = `
Object {
  "blockText": "",
  "layerPath": Array [],
  "overrides": Array [
    Object {
      "_class": "overrideValue",
      "overrideName": "d6593501-b089-4390-bbe2-fb10afb5df5a/1dea1c4d-f1bd-473b-a1aa-a0c6a1481ae2_blockText",
      "value": "",
    },
    Object {
      "_class": "overrideValue",
      "overrideName": "d6593501-b089-4390-bbe2-fb10afb5df5a/ef2d5b26-aa1c-40d3-8bab-37c10bccc5cb_blockText",
      "value": "",
    },
    Object {
      "_class": "overrideValue",
      "overrideName": "aa722c35-9ba4-4bf3-a5d0-f7d17f02c361_blockText",
      "value": "hello",
    },
    Object {
      "_class": "overrideValue",
      "overrideName": "83d2fdeb-6f4d-4948-a677-fe1f2aac64d5/6b386c69-d6cf-4c2f-ae06-c92af43268d5_blockText",
      "value": "",
    },
    Object {
      "_class": "overrideValue",
      "overrideName": "83d2fdeb-6f4d-4948-a677-fe1f2aac64d5/eee85c94-7361-4bcf-8afb-f59c6e8661f7_blockText",
      "value": "",
    },
  ],
  "symbolId": "61eb943b-7ee7-4b33-9315-aac7cb2d2764",
}
`;

exports[`serializes block with layer text: nodes 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "text": "",
      },
    ],
    "isRoot": true,
    "layerPath": Array [],
    "symbolId": "61eb943b-7ee7-4b33-9315-aac7cb2d2764",
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "text": "",
      },
    ],
    "isRoot": false,
    "layerPath": Array [
      "d6593501-b089-4390-bbe2-fb10afb5df5a",
      "1dea1c4d-f1bd-473b-a1aa-a0c6a1481ae2",
    ],
    "symbolId": "a1f7266f-50cc-416f-9ed2-4af4bca30257",
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "text": "",
      },
    ],
    "isRoot": false,
    "layerPath": Array [
      "d6593501-b089-4390-bbe2-fb10afb5df5a",
      "ef2d5b26-aa1c-40d3-8bab-37c10bccc5cb",
    ],
    "symbolId": "955f4cb1-7879-4331-a359-a21a031147cb",
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "text": "hello",
      },
    ],
    "isRoot": false,
    "layerPath": Array [
      "aa722c35-9ba4-4bf3-a5d0-f7d17f02c361",
    ],
    "symbolId": "b8441949-12ba-42be-8f87-5cf222af70d0",
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "text": "",
      },
    ],
    "isRoot": false,
    "layerPath": Array [
      "83d2fdeb-6f4d-4948-a677-fe1f2aac64d5",
      "6b386c69-d6cf-4c2f-ae06-c92af43268d5",
    ],
    "symbolId": "cdf2346b-cb21-4f23-8d93-7c4fb2e3a5a0",
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "text": "",
      },
    ],
    "isRoot": false,
    "layerPath": Array [
      "83d2fdeb-6f4d-4948-a677-fe1f2aac64d5",
      "eee85c94-7361-4bcf-8afb-f59c6e8661f7",
    ],
    "symbolId": "5be80b87-7bd6-4090-8947-316d54064544",
    "type": "paragraph",
  },
]
`;

exports[`serializes block with layer text: originalContent 1`] = `
Object {
  "blockText": "",
  "layerPath": Array [],
  "overrides": Array [
    Object {
      "_class": "overrideValue",
      "overrideName": "aa722c35-9ba4-4bf3-a5d0-f7d17f02c361_blockText",
      "value": "hello",
    },
  ],
  "symbolId": "61eb943b-7ee7-4b33-9315-aac7cb2d2764",
}
`;
